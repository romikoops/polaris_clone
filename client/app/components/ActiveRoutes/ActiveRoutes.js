import React, {Component} from 'react';
// import {EmailSignInForm} from 'redux-auth/bootstrap-theme';
import './ActiveRoutes.scss';
import PropTypes from 'prop-types';
// import SignIn from '../SignIn/SignIn';
export class ActiveRoutes extends Component {
    render() {
        const activeRoutesData = [
            {
                name: 'New York',
                country: 'USA',
                image: 'https://s3.eu-central-1.amazonaws.com/imcdev/assets/images/welcome/country/NY.jpg'
            },
            {
                name: 'Shanghai',
                country: 'China',
                image: 'https://s3.eu-central-1.amazonaws.com/imcdev/assets/images/welcome/country/shanghai.jpg'
            },
            {
                name: 'Singapore',
                country: 'Singapore',
                image: 'https://s3.eu-central-1.amazonaws.com/imcdev/assets/images/welcome/country/Singapore.jpg'
            },
            {
                name: 'Seoul',
                country: 'South Korea',
                image: 'https://s3.eu-central-1.amazonaws.com/imcdev/assets/images/welcome/country/seoul.jpg'
            },
            {
                name: 'Hanoi',
                country: 'Vietnam',
                image: 'https://s3.eu-central-1.amazonaws.com/imcdev/assets/images/welcome/country/Hanoi.jpg'
            },
            {
                name: 'Shenzhen',
                country: 'China',
                image: 'https://s3.eu-central-1.amazonaws.com/imcdev/assets/images/welcome/country/Shenzhen.jpg'
            }
        ];
        const theme = this.props.tenant.theme;
        const activeRouteBoxes = [];
        activeRoutesData.map((route, index) => {
            let divStyle = {
                backgroundImage: 'url(' + route.image + ')',
            };
            let textStyle = {
                background: theme && theme.colors ? '-webkit-linear-gradient(left, ' + theme.colors.primary + ',' + theme.colors.secondary + ')' : 'black'
            };

            const arb = (<div key={index} className="active_route flex-33 layout-row layout-align-center-center" style={divStyle}>
                <div className="flex-none layout-column layout-align-center-center">
                  <h2 className="city flex-none" style={textStyle} > {route.name} </h2>
                  <h5 className="country flex-none"> {route.name} </h5>
                </div>
            </div>);
            activeRouteBoxes.push(arb);
        });
        return (
      <div className="layout-row flex-100 layout-wrap active_routes">
        <div className="service_label layout-row layout-align-center-center flex-100">
            <h2 className="flex-none"> Active LCL Routes </h2>
        </div>
        {activeRouteBoxes}
      </div>
    );
    }
}

ActiveRoutes.propTypes = {
    tenant: PropTypes.object
};
