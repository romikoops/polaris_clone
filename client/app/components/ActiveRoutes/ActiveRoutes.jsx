import React from 'react'
import { withNamespaces } from 'react-i18next'
import styles from './ActiveRoutes.scss'
import PropTypes from '../../prop-types'
import { Carousel } from '../Carousel/Carousel'
import { activeRoutesData } from '../../constants'

const actRoutesData = activeRoutesData

export function ActiveRoutes ({ theme, t }) {
  return (
    <div className={`layout-row flex-100 layout-wrap ${styles.active_routes}`}>
      <div className={`${styles.service_label} layout-row layout-align-center-center flex-100`}>
        <h2 className="flex-none">{t('common:availableDestinations')}</h2>
      </div>
      <Carousel theme={theme} slides={actRoutesData} noSlides={4} />
    </div>
  )
}

ActiveRoutes.propTypes = {
  theme: PropTypes.theme,
  t: PropTypes.func.isRequired
}

ActiveRoutes.defaultProps = {
  theme: null
}

export default withNamespaces('common')(ActiveRoutes)
