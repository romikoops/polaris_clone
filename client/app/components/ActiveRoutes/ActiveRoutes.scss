@import "../../styles/defaults";
 :local {
    .active_routes {
      height: max-content;
      box-shadow: 0 2px 5px rgba(black, .5);
      position: relative;
    }
    .active_route{
      height: 300px;
      width: 350px;
      background-position: center;
      background-size: cover;
      background-repeat: no-repeat;
      position: relative;
    }
    .city {
      position: absolute;
      top: 20px;
      left: 58px;
      color: white;
      font-size: 40px;
      text-shadow: 0 1px 16px rgba(black,0.7);
      font-weight: normal;
    }
    .country {
      position: absolute;
      top: 70px;
      left: 58px;
      color: white;
      font-size: 24px;
      text-shadow: 0 2px 9px rgba(black,0.8);
    }
    .service_label{
      background-color: $light-grey;
      padding: 70px 0 40px 0;
      h2 {
        font-weight: normal;
      }
    }
    .fade {
      position: absolute;
      height: 100%;
      width: 100%;
      top: 0;
      right:0;
      background: rgba(1,12,13,0.45);
    }
    .slick_slide_ie11 {
      height: 350px;
      width: 450px;
      max-width: 100%;
      position: relative;
      background-position: center;
      background-size: cover;
      background-repeat: no-repeat;
      overflow-y: hidden; 
      overflow-x: hidden;
      padding-right: 20px;
    }
    .slick_slide {
      height: 350px;
      width: 450px;
      position: relative;
      .inner_wrapper {
        height: 100%;
        background-position: center;
        background-size: cover;
        background-repeat: no-repeat;
        min-width: 1px !important;
      }
    }
    .one_slide .slick_slide {
      // width: 1200px !important;
      // max-width: 1200px !important;
    }
    .slick_content {
      position: absolute;
      height: 100%;
      min-height: 350px;
      width: 100%;
      background-position: center;
      background-size: cover;
      background-repeat: no-repeat;
    }
    .slick_city {
      position: absolute;
      top: 20px;
      left: 58px;
      color: white;
      font-size: 40px;
      text-shadow: 0 1px 16px rgba(black,0.7);
      font-weight: normal;
    }
    .slick_country {
      position: absolute;
      top: 70px;
      left: 58px;
      color: white;
      font-size: 24px;
      text-shadow: 0 2px 9px rgba(black,0.8);
    }
    .slider_container{
      width: 100%;
      height: 350px;

      *:not(.slick-list) {
        min-height: 0;
        min-width: 0;
        // height: 100%;
      }
      
    }
  }
