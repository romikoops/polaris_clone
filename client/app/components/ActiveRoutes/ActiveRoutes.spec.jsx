import * as React from 'react'
import { mount } from 'enzyme'

import { ActiveRoutes } from './ActiveRoutes'

test('has multiple div and h2', () => {
  const props = {
    text: 'foo'
  }
  const wrapper = mount(<ActiveRoutes {...props} />)

  expect(wrapper.find('div').length).toBeGreaterThan(30)
  expect(wrapper.find('h2').length).toBeGreaterThan(10)
})

test('its text content contains specific parts', () => {
  const props = {}
  const wrapper = mount(<ActiveRoutes {...props} />)
  const text = wrapper.text()

  const expectedParts = [
    'Available',
    'Destinations',
    'China'
  ]

  expectedParts.forEach((singlePart) => {
    expect(text.includes(singlePart)).toBeTruthy()
  })
})
