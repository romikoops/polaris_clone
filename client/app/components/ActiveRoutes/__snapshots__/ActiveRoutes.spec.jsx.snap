// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow 1`] = `
<div
  className="layout-row flex-100 layout-wrap active_routes"
>
  <div
    className="service_label layout-row layout-align-center-center flex-100"
  >
    <h2
      className="flex-none"
    >
      Available Destinations
    </h2>
  </div>
  <Carousel
    fade={false}
    noSlides={4}
    slides={
      Array [
        Object {
          "header": "New York",
          "image": "https://assets.itsmycargo.com/assets/cityimages/NY_sm.jpg",
          "subheader": "USA",
        },
        Object {
          "header": "Shanghai",
          "image": "https://assets.itsmycargo.com/assets/cityimages/shanghai_sm.jpg",
          "subheader": "China",
        },
        Object {
          "header": "Singapore",
          "image": "https://assets.itsmycargo.com/assets/cityimages/Singapore_sm.jpg",
          "subheader": "Singapore",
        },
        Object {
          "header": "Seoul",
          "image": "https://assets.itsmycargo.com/assets/cityimages/seoul_sm.jpg",
          "subheader": "South Korea",
        },
        Object {
          "header": "Hanoi",
          "image": "https://assets.itsmycargo.com/assets/cityimages/Hanoi_sm.jpg",
          "subheader": "Vietnam",
        },
        Object {
          "header": "Shenzhen",
          "image": "https://assets.itsmycargo.com/assets/cityimages/Shenzhen_sm.jpg",
          "subheader": "China",
        },
      ]
    }
    theme={null}
  />
</div>
`;
